ipa-core/src/net/client/mod.rs

```diff
@@ -153,39 +155,15 @@ impl Future for ResponseFuture {
 /// TODO: It probably isn't necessary to always use `[MpcHelperClient; 3]`. Instead, a single
 ///       client can be configured to talk to all three helpers.
 #[derive(Debug, Clone)]
-pub struct MpcHelperClient {
+pub struct MpcHelperClient<R: TransportRestriction = Ring> {
```

Comment:
Maybe `ConnectionFlavor` or `Connectivity` are better names for it. Also we may need a follow up to remove the default value - these tend to produce cryptic compile errors when generics are omitted

Reply:
I spent some time thinking about this name and it's a tricky one.

I don't think `Connectivity` conveys enough information. I can go with `ConnectionFlavor` if you think its better. In [the Sharding article in Wikipedia](https://en.wikipedia.org/wiki/Shard_(database_architecture)) they refer to vertical and horizontal as "Partitions". So maybe `ConnectionPartition` ?

Reply:
Maybe @andyleiserson can chime in too - naming is hard. I don't think MPC connection can be classified as partitions, so `ConnectionFlavor` looks reasonable to me